lib/cache/repositories/impressions/redis_repository.rb

```diff
@@ -39,7 +39,7 @@ def add_bulk(key, bucketing_key, treatments_labels_change_numbers, time)
           # delete fetched impressions afterwards
           def clear
             impressions = impression_keys.each_with_object([]) do |key, memo|
-              _, _, ip, _ = key.split('/')
+              ip = key.split('/')[-2]
```

Comment:
add still a comment of a sample string you are parsing for the reviewers and for the posterity.